src/ActionsMenu.js

```diff
@@ -0,0 +1,141 @@
+// @flow
+import React from 'react';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import type {IntlShape} from 'react-intl';
+
+import ActionsMenuToggle from './ActionsMenuToggle';
+import ActionsMenuItem from './ActionsMenuItem';
+
+import './ActionsMenu.scss';
+
+import type {ActionToggleRegister} from './types';
+
+type ActionsMenuProps = {
+    intl: IntlShape,
+    changeHandler?: (event: Event, commandName: string) => void,
+    editingDisabled?: boolean,
+    // TODO: Flesh this definition out.
+    menuItems: {},
+    allowedActions: ActionToggleRegister,
+    usedActions: ActionToggleRegister
+};
+
+type ActionsMenuState = {
+    showMenu: boolean
+};
+
+class ActionsMenu extends React.Component<ActionsMenuProps, ActionsMenuState> {
+    static defaultProps = {
+        changeHandler: () => {},
+        editingDisabled: false,
+        usedActions: {},
+        menuItems: {
+            forward1: {
+                isAllowed: true,
+                labelKey: "Command.forward1"
+            },
+            forward2: {
+                isAllowed: true,
+                labelKey: "Command.forward2"
+            },
+            forward3: {
+                isAllowed: false,
+                labelKey: "Command.forward3"
+            },
+            left45: {
+                isAllowed: true,
+                labelKey: "Command.left45"
+            },
+            left90: {
+                isAllowed: true,
+                labelKey: "Command.left90"
+            },
+            left180: {
+                isAllowed: false,
+                labelKey: "Command.left180"
+            },
+            right45: {
+                isAllowed: true,
+                labelKey: "Command.right45"
+            },
+            right90: {
+                isAllowed: true,
+                labelKey: "Command.right90"
+            },
+            right180: {
+                isAllowed: false,
+                labelKey: "Command.right180"
+            }
+        }
+    }
+    constructor (props: ActionsMenuProps) {
+        super(props);
+        this.state = { showMenu: false };
+    }
+
+    render() {
+        return (
+            <React.Fragment>
+                <div className='ActionsMenu__header'>
+                    <h2 className='ActionsMenu__header-heading'>
+                        <FormattedMessage id='ActionsMenu.title' />
+                    </h2>
+                    <ActionsMenuToggle
+                        className='ActionsMenu__header-toggle'
+                        intl={this.props.intl}
+                        editingDisabled={!!this.props.editingDisabled}
+                        onClick={this.showHideMenu}
```

Comment:
Using showHideMenu for onClick creates coupling between keyboard handler and click handler. Even though it seems redundant, but I would suggest to implement a separate handler for click event, that calls showHideMenu.

Reply:
I would also change this property name from onClick to handleShowHideMenu, as this property is used to implement keyboard handler functions in ActionsMenuToggle component, and should be used to implement click handler in ActionsMenuToggle.